tests/infrastructure/golden_images/update_boot_source/test_ssp_common_templates_boot_sources.py

```diff
@@ -21,25 +21,25 @@
     get_artifactory_secret,
     validate_os_info_vmi_vs_linux_os,
 )
-from utilities.virt import VirtualMachineForTestsFromTemplate, running_vm
+from utilities.storage import data_volume_template_with_source_ref_dict
+from utilities.virt import VirtualMachineForTests, VirtualMachineForTestsFromTemplate, running_vm
 
 LOGGER = logging.getLogger(__name__)
 RHEL9_NAME = "rhel9"
 
 
-def assert_os_version_mismatch_in_vm(vm, expected_os):
-    expected_os_params = re.match(r"(?P<os_name>[a-z]+)(-stream)?(?P<os_ver>[0-9]+)", expected_os).groupdict()
+def assert_os_version_mismatch_in_vm(vm, latest_fedora_release_version):
```

Comment:
If you want to cover instance types, that is great. But let's not remove coverage for templates till they are removed.

Reply:
+1

Reply:
should we cover both?

Reply:
I see your point, but I respectfully disagree that this test is meant to cover templates. Its primary focus is on the Data Import Cron, and while it does use a template, that’s more of a tool rather than the core intent of the test.

In the past, creating a VM from templates was a common practice, and as a result, VirtualMachineForTemplates became the default approach. However, changing the method here does not necessarily imply a loss of coverage. Let me know what you think!

Reply:
The test file is called "test_ssp_common_templates_boot_sources.py". I am not in favor of removing template coverage with instance type as of now. When templates are removed, please feel free to remove associated tests. Till then if you want to add coverage for instance type, please add new tests in a separate file.